### PR TITLE
feat: [Speech] Adds a profanity filter sample

### PR DESCRIPTION
**Pull request by b-loved-dreamer**
I am adding a simple sample that showcases the use of the profanity filter feature.

### CODE REVIEW

File: speech/api/Recognize/Recognize.cs [hunk 1]
```diff
@@ -443,6 +446,29 @@ static object SyncRecognizeGcs(string storageUri)
...
         // [END speech_transcribe_sync_gcs]
 
+        // [START speech_transcribe_with_profanity_filter_sync_gcs]
+        static object SyncRecognizeWithProfanityFilterGcs(string storageUri)
+        {
+            var speech = SpeechClient.Create();
+            var response = speech.Recognize(new RecognitionConfig()
+            {
+                Encoding = RecognitionConfig.Types.AudioEncoding.Linear16,
+                SampleRateHertz = 16000,
```
Comment (laljikanjareeya) [inline speech/api/Recognize/Recognize.cs:456]: @b-loved-dreamer  This one is throwing and below Exception while running test Please take a look:
` Error: Grpc.Core.RpcException : Status(StatusCode=InvalidArgument, Detail="sample_rate_hertz (16000) in RecognitionConfig must either be omitted or match the value in the WAV header ( 8000).") `

**Review by laljikanjareeya — CHANGES_REQUESTED** [review]

File: speech/api/Recognize/Recognize.cs (hunk 1 above)
Comment (b-loved-dreamer) [reply speech/api/Recognize/Recognize.cs:456]: @laljikanjareeya thank you for reviewing. I will take a look.

**Comment by b-loved-dreamer** [comment]
Hi @amanda-tarafa, @jskeet; can you please take a look at this PR and stamp it for me. Thank you!

File: speech/api/Recognize/Recognize.cs [hunk 2]
```diff
@@ -443,6 +446,29 @@ static object SyncRecognizeGcs(string storageUri)
         }
         // [END speech_transcribe_sync_gcs]
 
+        // [START speech_transcribe_with_profanity_filter_sync_gcs]
+        static object SyncRecognizeWithProfanityFilterGcs(string storageUri)
+        {
+            var speech = SpeechClient.Create();
+            var response = speech.Recognize(new RecognitionConfig()
```
Comment (amanda-tarafa) [inline speech/api/Recognize/Recognize.cs:453]:
```suggestion
            var response = speech.Recognize(new RecognitionConfig
```

File: speech/api/Recognize/Recognize.cs [hunk 3]
```diff
@@ -674,6 +700,7 @@ public static int Main(string[] args)
                 >(args).MapResult(
                 (SyncOptions opts) => IsStorageUri(opts.FilePath) ?
                     SyncRecognizeGcs(opts.FilePath) : opts.EnableWordTimeOffsets ?
+                    SyncRecognizeWithProfanityFilterGcs(opts.FilePath) : opts.EnableProfanityFilter ?
                     SyncRecognizeWords(opts.FilePath) : opts.EnableAutomaticPunctuation ?
```
Comment (amanda-tarafa) [inline speech/api/Recognize/Recognize.cs:704]: The way you have it, `SyncRecognizeWithProfanityFilterGcs` will execute if `EnableWordTimeOffsets` and `SyncRecognizeWords` will execute if `EnableProfanityFilter`, I belive you want the other way around, like this:

```suggestion
                    SyncRecognizeGcs(opts.FilePath) : opts.EnableWordTimeOffsets ?
                    SyncRecognizeWords(opts.FilePath) : opts.EnableProfanityFilter ?
                    SyncRecognizeWithProfanityFilterGcs(opts.FilePath) : opts.EnableAutomaticPunctuation ?
```

File: speech/api/SpeechTest/SpeechTest.cs [hunk 4]
```diff
@@ -74,6 +74,7 @@ public abstract class CommonRecognizeTests
         protected string _audioFlacPath = Path.Combine("resources", "audio.flac");
         protected string _audioWavPath = Path.Combine("resources", "commercial_mono.wav");
         protected string _audioSteroPath = Path.Combine("resources", "commercial_stereo.wav");
+        protected static readonly string s_AUDIO_FILE = "gs://cloud-samples-data/speech/commercial_mono.wav";
```
Comment (amanda-tarafa) [inline speech/api/SpeechTest/SpeechTest.cs:77]:
```suggestion
        protected const string AudioFilePath = "gs://cloud-samples-data/speech/commercial_mono.wav";
```

File: speech/api/SpeechTest/SpeechTest.cs [hunk 5]
```diff
@@ -140,10 +141,10 @@ public void TestWithContext()
         [Fact]
         public void TestSyncWords()
         {
-            var output = Run("sync", "-w", _audioRawPath);
+            var output = Run("sync", "-w", s_AUDIO_FILE);
```
Comment (amanda-tarafa) [inline speech/api/SpeechTest/SpeechTest.cs:144]:
```suggestion
            var output = Run("sync", "-w", AudioFilePath);
```

File: speech/api/SpeechTest/SpeechTest.cs [hunk 6]
```diff
@@ -140,10 +141,10 @@ public void TestWithContext()
...
             Assert.Equal(0, output.ExitCode);
-            Assert.Contains("Brooklyn", output.Stdout);
-            Assert.Contains("WordStartTime:", output.Stdout);
+            Assert.Contains("Chromecast", output.Stdout);
+            Assert.Contains("Chromecast", output.Stdout);
```
Comment (amanda-tarafa) [inline speech/api/SpeechTest/SpeechTest.cs:147]: Don't need to assert the same thing twice
```suggestion
            Assert.Contains("Chromecast", output.Stdout);
```

File: speech/api/SpeechTest/SpeechTest.cs (hunk 5 above)
Comment (amanda-tarafa) [inline speech/api/SpeechTest/SpeechTest.cs:144]: If `_audioRawPath`, the one that was being used here before, is not being used anymore, please delete the variable declaration and the audio file as part of this PR.

File: speech/api/SpeechTest/SpeechTest.cs [hunk 7]
```diff
@@ -194,6 +195,14 @@ public void TestSyncRecognitionMetadata()
             Assert.Equal(0, output.ExitCode);
             Assert.Contains("Brooklyn", output.Stdout);
         }
+
+        [Fact]
+        public void TestSyncRecognitionProfanityFilter()
+        {
+            var output = Run("sync", "-f", s_AUDIO_FILE);
```
Comment (amanda-tarafa) [inline speech/api/SpeechTest/SpeechTest.cs:202]:
```suggestion
            var output = Run("sync", "-f", AudioFilePath);
```

**Review by amanda-tarafa — CHANGES_REQUESTED** [review]
A few changes requested. And also, this is not a chore, is a feature, addind a new sample in the samples repo is a feature.
Thanks.

**Review by amanda-tarafa — APPROVED** [review]
LGTM

**Comment by amanda-tarafa** [comment]
Let me know when this is ready to merge, I'll have to do it, we usually we require 2 reviews, one from a language owner (that'd be me) and another for a product owner, which we don't seem to have for Speech.
Thanks!

**Comment by b-loved-dreamer** [comment]
My update to this repo is completed. I am going to be adding a new sample in the coming weeks. In my view, this is ready to be merged. Maybe @jskeet can review this for us.

**Comment by jskeet** [comment]
No, @amanda-tarafa has already said that she's reviewed it from the language expert side of things, which is the review I'd be doing. We need someone from the Speech team to look from the product side.

**Comment by b-loved-dreamer** [comment]
Thanks for clarifying. I am unsure if there's anything I can do from my end. @amanda-tarafa says she's unaware of any product owner for Speech.

**Comment by amanda-tarafa** [comment]
I've usually merged Speech myself. I'll be doing so now.

**Pull request merged**